libraries/freertos_plus/aws/greengrass/test/aws_test_greengrass_discovery.c

```diff
@@ -1004,12 +1023,17 @@ TEST( Full_GGD, JSONRequestGetFile )
         if( xStatus == pdPASS )
         {
             ulByteRead = 0;
-            xStatus = GGD_JSONRequestGetFile( &xSocket,
-                                              cBuffer,
-                                              ulBufferSize,
-                                              &ulByteRead,
-                                              &xJSONFileRetrieveCompleted,
-                                              ulJSONFileSize - 1 ); /* Remove one byte to the expected JSON file size. */
+            ulRequestLoopCounter = 0;
+
+            do
+            {
+                xStatus = GGD_JSONRequestGetFile( &xSocket,
```

Comment:
Two questions:

1. This logic looks a little different than the other two blocks, but they seem to be doing the same thing.  Is there any reason for the difference?
1. Since we have 3 blocks of code that look almost the same, could this operation be moved to a static utility function?

Reply:
1. Good catch, it should be similar to the other three.
2. Created a static helper function per your suggestion.